docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,98 @@
+
+> It should also be noted here that, regardless of the choice made, we may need to alter how app specs are made, interpreted, and rendered. The bulk import design has various inputs and outputs listed in horizontal rows while the existing app specs are rendered vertically. There's no notion of rows/columns in the app spec right now or how to visually display various inputs along a grid pattern. In order to effectively match the design we may need to do one of the following:
+> 1. Alter the app specs to support multiple in-line parameters while retaining backward compatibility (including documentation, etc.).
+> 2. Use a distinct app spec for importers that can include these options.
+> 3. Hard-code how all import apps look and feel in the Narrative interface itself.
+>
+> If we decide to update how app specs work, that means working in the `narrative_method_store` and `catalog` repos as well as internal documentation.
+
+## Alternatives Considered
+
+* Implement the UI in the existing App Cell codebase
+* Implement the UI in a new Cell type
+* Don't use a cell, but some other interface
+
+## Decision Outcome
+
+## Consequences
+
+## Pros and Cons of the Alternatives
+
+### Implement the UI in the existing App Cell code base
+
+This option would entail adding a different mode for the current App Cell, likely based on some additional metadata feature.
+
+* `+` Reduce spread of additional, yet similar, cell types with similar code.
+* `+` Reuse existing cell control mechanisms currently in place with minimal modification.
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
+
+### Implement the UI in a new Cell type
+
+This option will create a new nbextension to serve a new cell type as an extension to the code cell.
+
+* `+` This would mean working with a fresh codebase without existing constraints.
+* `+` Intermediate, minimally functional changes, can be rapidly prototyped and released.
+* `-` To stick with the DRY principle, this would mean refactoring parts of the current app cell to make various components importable (message bus responses, job state management, tab menu, etc.) instead of their integration (really we might want to consider that anyway).
+* `-` Much of the structure of the mockup already functions similar to the existing App Cell and might become duplicated work with additional maintenance burden.
```

Comment:
these two somewhat cancel each other out -- presumably if we were to create a new cell type, shared functionality would be abstracted out in some way so that we didn't end up with a copy-pasta codebase

Reply:
(this is just a minor observation, no action needed)